submission/distributor.go

```diff
@@ -30,8 +30,22 @@ import (
 	"github.com/google/certificate-transparency-go/x509"
 
 	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/trillian/monitoring"
 )
 
+var (
+	// Metrics are per-log and add per-endpoint.
+	once        sync.Once
+	reqsCounter monitoring.Counter   // logURL, ep => value
+	rspLatency  monitoring.Histogram // logURL, ep => value
+)
+
+// initMetrics initializes all the exported metrics.
+func initMetrics(mf monitoring.MetricFactory) {
+	reqsCounter = mf.NewCounter("http_reqs", "Number of requests", "logid", "ep")
+	rspLatency = mf.NewHistogram("http_latency", "Latency of responses in seconds", "logid", "ep")
```

Comment:
Maybe `logurl` rather than `logid` (we use `logid` for the (numeric) tree ID in CTFE)?

Reply:
Updated.

Reply:
Ah, sorry, I wasn't clear -- the CTFE (and internal bits of Trillian) have metrics that are labelled by `logid` (or sometimes `log_id`) and this holds the Trillian tree ID.  That number is internal to the CT log and so is not accessible here, but I thought it might be nice to avoid using the same metric label.

The RFC-6962-defined log ID is a hash of a public key, and is much less user-friendly than a log URL or log name.

Reply:
ok, moved back to url. Updated description.